### PR TITLE
Magnetic ordering and deformation workflows

### PR DESCRIPTION
**Pull request by mkhorton**
## Summary

See #181 and #266 for discussion.

@computron this is a clean branch with some extra tests added and the previous specific comments addressed, happy to discuss the broader comments during our call later.

### CODE REVIEW

File: atomate/vasp/workflows/base/magnetism.py [hunk 1]
```diff
@@ -0,0 +1,871 @@
...
+                len(SpaceGroup.from_int_number(n).symmetry_ops)
+                for n in symmetry_int_numbers
+            ]
+
+            # find the largest values...
+            max_symmetries = sorted(list(set(num_sym_ops)), reverse=True)
+
+            # ...and decide which ones to keep
+            if len(max_symmetries) > self.truncate_by_symmetry:
+                max_symmetries = max_symmetries[0:5]
```
Comment (computron) [inline atomate/vasp/workflows/base/magnetism.py]: why is this ```0:5``` and not ```0:self.truncate_by_symmetry```

**Comment by mkhorton** [comment]
Thanks for the additional comments; I do have a fork where I'm migrating some of this over to pymatgen, will keep you updated.

File: atomate/vasp/workflows/base/magnetism.py (hunk 1 above)
Comment (mkhorton) [reply atomate/vasp/workflows/base/magnetism.py]: Thanks for catching this!

Comment (mkhorton) [reply atomate/vasp/workflows/base/magnetism.py]: This is consistent with the docstring, but agreed would make more sense to be able to set this arbitrarily.

**Comment by mkhorton** [comment]
@computron moved a lot of functionality from this PR to pymatgen now (-542 loc) in response to your comments

File: atomate/vasp/workflows/base/magnetism.py [hunk 2]
```diff
@@ -0,0 +1,871 @@
...
+
+        # ... and strip out existing magmoms, which can cause conflicts
+        # with later transformations otherwise since sites would end up
+        # with both magmom site properties and Specie spins defined
+        if "magmom" in input_structure.site_properties:
+            input_structure.remove_site_property("magmom")
+
+        return input_structure
+
+    def _generate_transformations(self, structure):
```
[the remark this answers is not in the document]
Comment (mkhorton) [reply atomate/vasp/workflows/base/magnetism.py]: Moved

File: atomate/vasp/workflows/base/magnetism.py [hunk 3]
```diff
@@ -0,0 +1,871 @@
...
+                mag_species_spin,
+                order_parameter=constraints,
+                **self.transformation_kwargs
+            )
+
+            transformations[name] = trans
+
+        return transformations
+
+    def _generate_ordered_structures(self, sanitized_input_structure, transformations):
```
[the remark this answers is not in the document]
Comment (mkhorton) [reply atomate/vasp/workflows/base/magnetism.py]: Moved

File: atomate/vasp/workflows/base/magnetism.py [hunk 4]
```diff
@@ -0,0 +1,871 @@
...
+                if idx not in structures_to_remove
+            ]
+            ordered_structures_origins = [
+                o
+                for idx, o in enumerate(ordered_structures_origins)
+                if idx not in structures_to_remove
+            ]
+
+        # also remove low symmetry structures
+        if self.truncate_by_symmetry:
```
[the remark this answers is not in the document]
Comment (mkhorton) [reply atomate/vasp/workflows/base/magnetism.py]: Agreed, but not addressing that in this present PR (there are other improvements we could make to Spacegroup also)

File: atomate/vasp/workflows/base/magnetism.py [hunk 5]
```diff
@@ -0,0 +1,871 @@
...
+            ordered_structures, ordered_structures_origins = _add_structures(
+                ordered_structures,
+                ordered_structures_origins,
+                structures_to_add,
+                origin=origin,
+            )
+
+        def _add_metadata(structure):
+            """
+            For book-keeping, store useful metadata with the Structure
```
[the remark this answers is not in the document]
Comment (mkhorton) [reply atomate/vasp/workflows/base/magnetism.py]: Added some more info here

**Pull request merged**